core/src/main/java/org/mule/runtime/core/internal/policy/CompositeSourcePolicy.java

```diff
@@ -125,7 +125,10 @@ public CompositeSourcePolicy(List<Policy> parameterizedPolicies,
    */
   @Override
   protected Publisher<CoreEvent> processPolicy(Policy policy, Processor nextProcessor, CoreEvent event) {
-    return just(event).transform(sourcePolicyProcessorFactory.createSourcePolicy(policy, nextProcessor));
+    logPolicy(getCoreEventId(event), getPolicyName(policy), getCoreEventAttributesAsString(event));
```

Comment:
this "before" log should be done inside the publisher chain returned by this method